Services/source/housekeeping/housekeeping_service.c

```diff
@@ -395,12 +437,19 @@ int num_digits(int num) {
   return count;
 }
 
-static inline void prv_get_lock(SemaphoreHandle_t *lock) {
+static inline Result prv_get_lock(SemaphoreHandle_t *lock) {
   if (*lock == NULL) {
     *lock = xSemaphoreCreateMutex();
+    if (*lock == NULL) {
+      return FAILURE;
+    }
   }
-  configASSERT(*lock);
-  xSemaphoreTake(*lock, portMAX_DELAY);
+  for (int i = 0; i < 10; i++){
+    if (xSemaphoreTake(*lock, portMAX_DELAY) == pdTRUE) {
```

Comment:
With a max_delay this function will only ever return true or not at all. Is having the Result return code necessary? Or do you intend on making this time out

Reply:
ah I see now. This change was made in my mess of trying to resolve bugs. I'll put it back how it ran before